### PR TITLE
feat(rails-patterns): add Rails framework patterns skill

### PR DESCRIPTION
**Pull request by rockwellwindsor**
## What Changed
Adds a rails-patterns framework skill covering Rails patterns: directory contract, skinny controllers with service objects, form objects, query objects, idiomatic ActiveRecord, background jobs, ViewComponent, Hotwire, and the Rails 8 Solid stack.

## Why This Change
Fills the Rails gap alongside the existing laravel-patterns and django-patterns framework skills. Defers decisions to rules/ruby/patterns.md and focuses on the how (code walkthroughs), per the rules-vs-skills split.

## Testing Done

- [ ] Manual testing completed
- [x] Automated tests pass locally (`node tests/run-all.js`)
- [ ] Edge cases considered and tested

## Type of Change
- [ ] `fix:` Bug fix
- [x] `feat:` New feature
- [ ] `refactor:` Code refactoring
- [ ] `docs:` Documentation
- [ ] `test:` Tests
- [ ] `chore:` Maintenance/tooling
- [ ] `ci:` CI/CD changes

## Security & Quality Checklist
- [x] No secrets or API keys committed (ghp_, sk-, AKIA, xoxb, xoxp patterns checked)
- [x] JSON files validate cleanly
- [ ] Shell scripts pass shellcheck (if applicable)
- [ ] Pre-commit hooks pass locally (if configured)
- [x] No sensitive data exposed in logs or output
- [x] Follows conventional commits format

## If you changed dependencies or `package.json` (`bin` / `files` / deps)
- [ ] Ran `yarn install --mode=update-lockfile` and committed the `yarn.lock` change. CI runs Yarn in hardened mode on public PRs and fails if the lockfile would be modified, so an out of date `yarn.lock` breaks the build even when nothing else is wrong.

## If you added a skill, command, agent, hook, or CLI tool
- [x] Registered in `package.json` (`bin` and `files`), `manifests/install-components.json`, `manifests/install-modules.json`, and `agent.yaml`
- [x] Regenerated the catalog (`npm run catalog:sync`) and command registry (`npm run command-registry:write`)
- [x] Updated the docs tables it belongs in (`README.md`, `COMMANDS-QUICK-REF.md`, `docs/COMMAND-AGENT-MAP.md`) - **README.md updated; command docs N/A.**
- [ ] If it ships a new script path, added it to the publish surface allowlist (`tests/scripts/npm-publish-surface.test.js`)
- [ ] Cross-harness surfaces updated if applicable (for Codex, `.agents/skills/<name>/` plus `agents/openai.yaml`; the Codex frontmatter validator allows only `name`, `description`, `metadata`, `license`, `allowed-tools`, so drop keys like `version` from that copy)
- [x] Full gauntlet passes locally (`npm test`)

## Documentation
- [x] Updated relevant documentation
- [ ] Added comments for complex logic
- [x] README updated (if needed)

### CODE REVIEW

File: skills/rails-patterns/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+class AccountingExportJob < ApplicationJob
+  queue_as :exports
+
+  retry_on AccountingApi::TransientError, wait: :polynomially_longer, attempts: 5
+  discard_on AccountingApi::PermanentError
+
+  def perform(invoice_id)
+    invoice = Invoice.find(invoice_id)
+    AccountingApi.export(invoice)
+  end
```
[the remark this answers is not in the document]
Comment (rockwellwindsor) [reply skills/rails-patterns/SKILL.md:302]: [834efe6](https://github.com/affaan-m/ECC/pull/3074/commits/834efe6409193f08ad00560200e59c4e73daa5be)

File: skills/rails-patterns/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,440 @@
+---
+name: rails-patterns
+description: Ruby on Rails framework patterns for Rails 7.1+ and 8.x apps. Covers the directory contract, skinny controllers with service objects, form objects, query objects, idiomatic ActiveRecord, background jobs, ViewComponent, Hotwire, and the Rails 8 Solid stack. Use when building or reviewing Rails apps, controllers, models, services, jobs, or views.
+origin: community
+---
+
+# Rails Patterns
+
+Framework patterns for modern Ruby on Rails applications (Rails 7.1+ and 8.x). Rails is opinionated by design; these are the patterns the community has converged on for apps that stay maintainable past the 50-model mark. This skill is the "how." For the "what" and "when" (the decisions about which pattern to reach for), see `rules/ruby/patterns.md`.
```
[the remark this answers is not in the document]
Comment (rockwellwindsor) [reply skills/rails-patterns/SKILL.md]: [1404414](https://github.com/affaan-m/ECC/pull/3074/commits/14044147124474958d90ecf488c93a3570fb6b8d)

File: skills/rails-patterns/SKILL.md [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+end
+```
+
+### The service object
+
+```ruby
+# app/services/invoices/create.rb
+module Invoices
+  class Create
+    Result = Data.define(:success?, :invoice, :errors)
```
[the remark this answers is not in the document]
Comment (rockwellwindsor) [reply skills/rails-patterns/SKILL.md]: bf41924

File: skills/rails-patterns/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+
+    def self.call(params:, user:) = new(params: params, user: user).call
```
[the remark this answers is not in the document]
Comment (rockwellwindsor) [reply skills/rails-patterns/SKILL.md]: bf41924

File: skills/rails-patterns/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,472 @@
...
+
+```ruby
+add_index :accounting_exports, :idempotency_key, unique: true
+```
+
+The unique index is what makes this safe: two concurrent attempts race on insert, one
+raises `ActiveRecord::RecordNotUnique`, and the retry finds the existing row. The guard
+covers the window before the remote call; passing `idempotency_key` through to the API
+covers the window after it, so a crash between the API call and `update!` still resolves
+to a single export.
```
[the remark this answers is not in the document]
Comment (rockwellwindsor) [reply skills/rails-patterns/SKILL.md]: [4c018a7](https://github.com/affaan-m/ECC/pull/3074/commits/4c018a76b70a9d0ee6e91992721a2a9b166612e3)

File: skills/rails-patterns/SKILL.md [hunk 6]
```diff
@@ -0,0 +1,476 @@
...
+```ruby
+add_index :accounting_exports, :idempotency_key, unique: true
+```
+
+The unique index is what makes this safe, and `create_or_find_by!` is what consumes it:
+it attempts the insert first, and when a concurrent attempt wins the race it rescues
+`ActiveRecord::RecordNotUnique` and loads the existing row itself. Active Record settles
+the race inside the call — no job-level retry is involved, and `retry_on` above covers
+only `AccountingApi::TransientError`. (`find_or_create_by!` selects before inserting, so
+it leaves that race open and can raise.) The guard
```
[the remark this answers is not in the document]
Comment (rockwellwindsor) [reply skills/rails-patterns/SKILL.md]: [c632872](https://github.com/affaan-m/ECC/pull/3074/commits/c632872a5291edca473929c5a18964ccef6d012d)

File: skills/rails-patterns/SKILL.md [hunk 7]
```diff
@@ -0,0 +1,476 @@
...
+end
+```
+
+```ruby
+add_column :posts, :comments_count, :integer, default: 0, null: false
+```
+
+`post.comments_count` becomes a column read instead of a `COUNT(*)`. On a table that
+already has rows, run `Post.reset_counters(id, :comments)` for each existing post after
+adding the column.
```
[the remark this answers is not in the document]
Comment (rockwellwindsor) [reply skills/rails-patterns/SKILL.md]: [925d4f0](https://github.com/affaan-m/ECC/pull/3074/commits/925d4f0513dae1550482da843fcab5e448556cec)

[the remark this answers is not in the document]
Comment (rockwellwindsor) [reply skills/rails-patterns/SKILL.md]: [925d4f0](https://github.com/affaan-m/ECC/pull/3074/commits/925d4f0513dae1550482da843fcab5e448556cec)
